### DISCUSSION TITLE
API

### DISCUSSION THREAD

**Discussion by ghost** [AzuraCast/AzuraCast#4556] [Q&A] [answered]
Hi,

Is there somebody who can help me with the correct API (JSON) post script to create a new Permission Role and assign the role to a Radio Station.
I had it working 100% once and completely forgot how I did it.
I am willing to exchange some $ for your help.

Kind Regards,

**Comment by ghost** [comment]
Anybody?

**Comment by BusterNeece** [comment, marked as answer]
You may have been encountering a bug with the `/api/admin/roles` endpoint not properly saving permissions; this should now be fixed, so you can send a `POST` to that endpoint with a body like:

```json
{
  name: "New Role Name",
  permissions: {
    global: ["administer all"],
    station: {
      1: ["administer all"]
    }
  }
}
```

And then assign that role (with the ID that's returned) to a user via `PUT /api/admin/user/(user_id)` (or `POST` to create a new one) with a value like `{ roles: [1] }`

**Comment by ghost** [comment]
Hi,

Thank you for taking the time to reply to my question on the Forums.
I have tried to create a new role by using POST and the following JSON code;

{
  "name": "New Role Name",
  "permissions": {
    "global": ["administer all"],
    "station": {
      "1": ["administer all"]
    }
  }
}

Although it does create the role without any problems, it still does not
save the Global permissions or save the assigned Station permissions.
I am using a brand new installation which runs under the Docker
environment. (Stable Release)(Ubuntu 20.04)

Kind Regards,

**Reply by BusterNeece** [reply]
The issue with saving permissions via the API call is only fixed on the latest Rolling Release version, not the latest Stable version just yet.

**Comment by ghost** [comment]
Hi,

I will do a new Rolling Release install and post the outcome here as
soon as it is done.
Thank you for a very well thought out product and giving us the
opportunity to use the API as a way to add value to our own Projects.

Kind Regards,

**Comment by ghost** [comment]
Hi,

I installed the Rolling Release version and it works 100% now.
Thank you very much for your time.

Kind Regards,
